src/PrestaShopBundle/Controller/Admin/MultistoreController.php

```diff
@@ -73,42 +75,51 @@ public function header(bool $lockedToAllShopContext): Response
             ]);
         }
 
-        $isAllShopContext = $this->multistoreContext->isAllShopContext();
-        $isShopContext = $this->multistoreContext->isShopContext();
-        $colorConfigLink = false;
-
-        if ($isShopContext) {
-            $currentContext = $this->entityManager->getRepository(Shop::class)->findOneBy(['id' => $this->multistoreContext->getContextShopID()]);
-            $colorConfigLink = $this->getAdminLink('AdminShop', ['shop_id' => $currentContext->getId(), 'updateshop' => true]);
-        } elseif (!$isAllShopContext) {
-            $shopGroupLegacy = $this->multistoreContext->getContextShopGroup();
-            $currentContext = $this->entityManager->getRepository(ShopGroup::class)->findOneBy(['id' => $shopGroupLegacy->id]);
-            $colorConfigLink = $this->getAdminLink('AdminShopGroup', ['id_shop_group' => $currentContext->getId(), 'updateshop_group' => true]);
-        } else {
-            // use ShopGroup object as a the container for "all shops" context so that it can be used transparently in twig
-            $currentContext = new ShopGroup();
-            $currentContext->setName($this->trans('All shops', 'Admin.Global'));
-            $currentContext->setColor('');
-        }
-
         $groupList = [];
         if (!$lockedToAllShopContext) {
             $groupList = $this->entityManager->getRepository(ShopGroup::class)->findBy(['active' => true]);
         }
 
-        $colorBrightnessCalculator = $this->get('prestashop.core.util.color_brightness_calculator');
-
-        return $this->render('@PrestaShop/Admin/Multistore/header.html.twig', [
-            'isMultistoreUsed' => $this->multistoreFeature->isUsed(),
-            'currentContext' => $currentContext,
+        return $this->renderHeader('@PrestaShop/Admin/Multistore/header.html.twig', [
             'groupList' => $groupList,
-            'isShopContext' => $isShopContext,
-            'link' => $this->getContext()->link,
-            'isTitleDark' => empty($currentContext->getColor()) ? true : $colorBrightnessCalculator->isBright($currentContext->getColor()),
-            'isAllShopContext' => $isAllShopContext,
-            'isGroupContext' => $this->multistoreContext->isGroupShopContext(),
             'lockedToAllShopContext' => $lockedToAllShopContext,
-            'colorConfigLink' => !$isAllShopContext && empty($currentContext->getColor()) ? $colorConfigLink : false,
+        ]);
+    }
+
+    /**
+     * This method returns a Response object containing the multistore header displayed at the top of product page
```

Comment:
Not sure if worth it but additional comments could be added

```suggestion
     * This method returns a Response object containing the multistore header displayed at the top of product page.
     * It is not designed to be called as usual controller action, therefore having no dedicated route configuration.
     * @see MultistoreComponentsExtension for its usage example.
```